### PR TITLE
[FIX] Reset`EarlyStopping` state in cross-validation when `refit=True`

### LINKED ISSUE #1124: How refit on train+val set?
How should I refit the model on the whole dataset containing both training and validation sets?

It seems that both `nf.fit` and `nf.cross_validation` don't support this feature, but AutoModels have `refit_with_val`. Could I set `refit_with_val` on non-auto models?

### PR DESCRIPTION
**Pull request by marcopeix**
Currently, each refit window inherits the `EarlyStopping` state. Since each window has a different validation set, the best score is stale and patience counter is not reset.

Now, `EarlyStopping` callback is reset for each consecutive fit.

### CODE REVIEW

File: neuralforecast/common/_base_model.py [hunk 1]
```diff
@@ -604,6 +604,18 @@ def _fit(
...
+        # Reset here so every fit starts with a fresh early-stopping counter.
+        for cb in self.trainer_kwargs.get("callbacks") or []:
+            if isinstance(cb, EarlyStopping):
+                cb.wait_count = 0
+                cb.stopped_epoch = 0
+                cb.best_score = (
+                    torch.tensor(torch.inf)
+                    if cb.mode == "min"
+                    else torch.tensor(-torch.inf)
+                )
```
Comment (gee-senbong) [inline neuralforecast/common/_base_model.py]: Any reason why not fixing this at the cross validation function? Would it be cleaner if we just construct a new `EarlyStopping` instance. By doing so, we don't need to handle these internal details manually. 

Unfortunately `pytorch_lightning` does not have `reset` method (https://docs.pytorch.org/ignite/generated/ignite.handlers.early_stopping.EarlyStopping.html#ignite.handlers.early_stopping.EarlyStopping.reset).

Comment (marcopeix) [reply neuralforecast/common/_base_model.py]: Don't want to do it just for cross-validation but for consecutive call to `fit` too. Which is why I don't fix it at the cross-validation function. But, I could rebuilt it from the supplied kwargs by the user. I think that's cleaner.

**Review by gee-senbong — APPROVED** [review]
